packages/devtools_app/lib/src/notifications.dart

```diff
@@ -235,19 +267,65 @@ class _NotificationState extends State<_Notification>
                 theme.primaryTextTheme.subtitle1,
             child: Padding(
               padding: const EdgeInsets.all(8.0),
-              child: Align(
-                alignment: Alignment.topLeft,
-                child: Text(
-                  widget.message,
-                  style: theme.textTheme.bodyText1,
-                  overflow: TextOverflow.visible,
-                  maxLines: 6,
-                ),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  _buildMessage(),
+                  const SizedBox(height: defaultSpacing),
+                  _buildActions(),
+                ],
               ),
             ),
           ),
         ),
       ),
     );
   }
+
+  Widget _buildMessage() {
+    return Text(
+      widget.message,
+      style: Theme.of(context).textTheme.bodyText1,
+      overflow: TextOverflow.visible,
+      maxLines: 6,
+    );
+  }
+
+  Widget _buildActions() {
+    if (widget.actions.isEmpty) return const SizedBox();
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        for (int i = 0; i < widget.actions.length; i++) ...[
```

Comment:
we should create a joinWith(List<Widget>, Widget separator) helper method or maybe extension method on List<Widget> as we've got this pattern a few places.